### ISSUE TITLE
WebIDL serializer has been deprecated in favor of toJSON operation

### ISSUE THREAD

**Issue by tobie** [w3c/resource-timing#112]
Hi!

We recently [deprecated WebIDL serializers][0]. You can now directly specify [toJSON operations][1] instead, which you previously weren't allowed to do.

To deal with common cases, we added a new [\[Default\] extended attribute][2] which triggers the [default toJSON operation][3] that behaves similarly to how `serializers={attributes}` or `serializers={attributes, inherit}` used to. That is, it serializes all attributes that are of a [JSON type][4] into a vanilla JSON object.

It seems the following interface in this spec is impacted by this change:

- [ ] [`PerformanceResourceTiming`](https://w3c.github.io/resource-timing/#idl-def-performanceresourcetiming)

It's a good candidate for the [default toJSON operation][3], so the below should be all you need, provided you also similarly update `PerformanceEntry` in its relevant specification:

```webidl
[Exposed=(Window,Worker)]
interface PerformanceResourceTiming : PerformanceEntry {
    readonly attribute DOMString initiatorType;
    // etc...
    [Default] object toJSON();
};
```

I'm sorry for the inconvenience this causes, but our hope is that this ultimately makes things a lot simpler and clearer for everybody.

Please feel free to reach out if you have any questions.

Thanks!

[0]: https://github.com/heycam/webidl/commit/a505f33
[1]: https://heycam.github.io/webidl/#idl-tojson-operation
[2]: https://heycam.github.io/webidl/#Default
[3]: https://heycam.github.io/webidl/#es-default-tojson
[4]: https://heycam.github.io/webidl/#dfn-json-types

**Comment by igrigorik** [comment]
@tobie was the dependency in respec already updated? I'm seeing new WebIDL failures on https://w3c.github.io/navigation-timing/ that are causing all the links and references to be broken.

**Comment by tobie** [comment]
Yes, @marcoscaceres updated Respec yesterday: https://github.com/w3c/respec/pull/1300.

Is there something we could do better next time so you don't feel like we've pulled the rung from under you?

**Comment by igrigorik** [comment]
Yikes, well.. this was a very short (and unspecified) deadline, which doesn't make things easy. We have a lot of folks referencing the github.io versions, which are both broken at the moment. IMO, breaking changes like this need to have a much longer lead time. 

@siusin @toddreifsteck any objections to us merging above toJSON changes?

**Comment by marcoscaceres** [comment]
Yeah, I should have added a warning. I figured this way we would just break things and quickly fix them.

**Comment by marcoscaceres** [comment]
(it only affects very few specs and it's a very easy fix - can send PR)

**Comment by igrigorik** [comment]
The affected specs are: HR-Time (https://github.com/w3c/hr-time/issues/49), Resource Timing (https://github.com/w3c/navigation-timing/issues/72), Navigation Timing. Any others I'm overlooking?

**Comment by tobie** [comment]
…and Performance Timeline (https://github.com/w3c/performance-timeline/issues/82).

**Comment by tobie** [comment]
Can't these specs just link to a previous version of Respec in the meantime?

**Comment by igrigorik** [comment]
Add Server-Timing to that list: https://w3c.github.io/server-timing/ 

@tobie yes, we could, but this is unnecessary pain -- I think we could have handled this a bit better.

**Comment by tobie** [comment]
>  I think we could have handled this a bit better.

I can't disagree. I'm just trying to help find a solution for now and try to figure out how this can be improved next time.

On that latter topic I've opened a bug here: https://github.com/heycam/webidl/issues/379.

**Comment by igrigorik** [comment]
Resolved via #114.
